### PR TITLE
Implement users.delete RPC function to replace DELETE /api/ui/v0/users/{userId}/ endpoint

### LINKED ISSUE #4000: Implement `users.delete` RPC function
We need to implement a `users.delete` RPC function to replace the `DELETE /api/ui/v0/users/{userId}/` endpoint.

The function signature should be:

```python
@rpc_method(name='users.delete')
@http_basic_auth_superuser_required
@handle_rpc_exceptions
def delete(
        *,
        user_id: int,
) -> None:
```

See [the wiki](https://wiki.mathesar.org/projects/2024/architecture-transition/rpc/#functions) for more context.

### PR DESCRIPTION
**Pull request by nilaygit-10721**
Fixes #4000 

This pull request implements the users.delete RPC function to replace the existing DELETE /api/ui/v0/users/{userId}/ endpoint. The function follows the specified signature and includes necessary decorators for authorization and error handling. This change enhances the API's structure by moving user model manipulations to the RPC layer, improving maintainability and scalability.

## Checklist
 
- [X] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [X] My pull request targets the `develop` branch of the repository
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: mathesar/api/rpc/users/base.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+from typing import Optional
+
+from django.contrib.auth import get_user_model
+from django.core.exceptions import ObjectDoesNotExist
+
+from mathesar.api.exceptions import APIError
```
Comment (mathemancer) [inline mathesar/api/rpc/users/base.py:6]: The `APIError` class is only for use in the (now deprecated) REST API. It's not a good idea here. You should raise a custom error class based on a standard Python `Exception` instead.

File: mathesar/api/rpc/users/base.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                )
+        
+        user.delete()
+        
+    except ObjectDoesNotExist:
+        raise APIError(
+            f"User with ID {user_id} not found.",
+            code="user_not_found",
+            status_code=404
+        )
```
Comment (mathemancer) [inline mathesar/api/rpc/users/base.py:43]: You don't need a custom error here. Just let it fail.

**Review by mathemancer — CHANGES_REQUESTED** [review]
Good start!

Please also:
- satisfy the linter
- add a couple tests (see the other RPC function tests)
- wire this up to the actual API by adding the appropriate line in the `common_settings.py` file
- add a docstring similar to the other RPC functions
- wire this into the project documentation in the `docs/docs/api/rpc.md` file.

Thanks for your effort so far!

**Comment by mathemancer** [comment]
Procedural note: In the future, please use the PR template.

**Comment by nilaygit-10721** [comment]
okay i will use pr template

**Pull request closed without merging**